labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise01.scala

```diff
@@ -99,7 +99,11 @@ object ListManipulationExercise01 {
    * Implement it whatever way suites you best. Hint: it can be done in a neat way using recursion. 
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    var lists = List(l)
+    while(l.length > 0) {
+      lists :: l.drop(1)
+    }
+    return lists
```

Comment:
Interesting, I didn't know that you could use :: to post-append (put `l.drop(1)` on the back).

I would also challenge you to think of a recursive approach.